modules/cookbooks/pages/kafka_migrator.adoc

```diff
@@ -0,0 +1,633 @@
+
+curl -X POST -u "kafka:testpass" -H "Content-Type: application/vnd.schemaregistry.v1+json" --data '{"schema": "{\"name\": \"Bar\", \"type\": \"record\", \"fields\": [{\"name\": \"data\", \"type\": \"int\"}]}"}' https://schema-registry-source.cloud.kafka.com:30081/subjects/bar/versions
+----
+
+endif::[]
+
+== Generate messages
+
+Let's simulate an application with a producer and consumer actively publishing and reading messages on the `source` cluster. You can use Redpanda Connect to generate some Avro-encoded messages and push them to the two topics from the `source` cluster.
+
+.`generate_data.yaml`
+
+ifndef::env-cloud[]
+
+[source,yaml]
+----
+http:
+  enabled: false
+
+input:
+  sequence:
+    inputs:
+      - generate:
+          mapping: |
+            let msg = counter()
+            root.data = $msg
+
+            meta kafka_topic = match $msg % 2 {
+              0 => "foo"
+              1 => "bar"
+            }
+          interval: 1s
+          count: 0
+          batch_size: 1
+
+        processors:
+          - schema_registry_encode:
+              url: "http://localhost:8081"
+              subject: ${! metadata("kafka_topic") }
+              avro_raw_json: true
+
+output:
+  kafka_franz:
+    seed_brokers: [ "localhost:9092" ]
+    topic: ${! @kafka_topic }
+    partitioner: manual
+    partition: ${! random_int(min:0, max:1) }
+----
+
+endif::[]
+
+ifdef::env-cloud[]
+
+[source,yaml]
+----
+http:
+  enabled: false
+
+input:
+  sequence:
+    inputs:
+      - generate:
+          mapping: |
+            let msg = counter()
+            root.data = $msg
+
+            meta kafka_topic = match $msg % 2 {
+              0 => "foo"
+              1 => "bar"
+            }
+          interval: 1s
+          count: 0
+          batch_size: 1
+
+        processors:
+          - schema_registry_encode:
+              url: "https://schema-registry-source.cloud.kafka.com:30081"
+              subject: ${! metadata("kafka_topic") }
+              avro_raw_json: true
+              basic_auth:
+                enabled: true
+                username: kafka
+                password: testpass
+
+output:
+  kafka_franz:
+    seed_brokers: [ "source.cloud.kafka.com:9092" ]
+    topic: ${! @kafka_topic }
+    partitioner: manual
+    partition: ${! random_int(min:0, max:1) }
+    tls:
+      enabled: true
+    sasl:
+      - mechanism: SCRAM-SHA-256
+        username: kafka
+        password: testpass
+----
+
+endif::[]
+
+Now, run this command to start the pipeline, and leave it running:
+
+[source,console]
+----
+redpanda-connect run generate_data.yaml
+----
+
+Next, add a Redpanda Connect consumer, which reads messages from the `source` cluster topics, and leave it running. This consumer uses the `foobar` consumer group, which will be reused in a later step when consuming from the `sink` cluster.
+
```

Comment:
You mention a `sink` cluster here but elsewhere a `destination`. Can you check?

Reply:
Nice catch! Copy / paste gone wrong :(